src/app/vendors/SingleVendor.tsx

```diff
@@ -165,14 +165,31 @@ function SingleVendor({
     return <ErrorPanel error={new Error(`No known vendor with hash ${vendorHash}`)} />;
   }
 
+  let displayName = d2Vendor.def.displayProperties.name;
+  let displayDesc = d2Vendor.def.displayProperties.description;
+
+  // if this vendor is the seasonal artifact
+  if (vendorDef.displayCategories.find((c) => c.identifier === 'category_reset')) {
+    // search for the associated item. this is way harder than it should be, but we have what we are given
+    const seasonHash = profileResponse?.profile.data?.currentSeasonHash;
+    const artifactDisplay = Object.values(defs.InventoryItem.getAll()).find(
+      //     it belongs to the current season,                         and looks like an artifact
+      (i) => i.seasonHash === seasonHash && i.inventory!.stackUniqueLabel.includes('.artifacts.')
+    )?.displayProperties;
+    if (artifactDisplay) {
+      displayName = artifactDisplay.name;
+      displayDesc = artifactDisplay.description;
+    }
+  }
+
```

Comment:
You could, theoretically, `useMemo` this, though I doubt it matters in practice.

Reply:
yeah i am banking on shockingly low use, just going for one fewer things that look like a DIM bug. i gate the big invitem loop and string matching behind cheaper checks